Framework/ScriptRepository/inc/MantidScriptRepository/ScriptRepositoryImpl.h

```diff
@@ -12,6 +12,12 @@
 #else
 #define SCRIPT_DLL_EXPORT DLLImport
 #endif
+#elif defined(__GNUC__) && !defined(__clang__)
```

Comment:
Can we move all of this to a `DllConfig.h` file to be consistent with everywhere else? This is all our own code and it would be better to be consistent.